nep-20.mediawiki

```diff
@@ -0,0 +1,533 @@
+    "InsufficientFunds" |
+    "ContentTypeNotSupported" |
+    "Error";
+
+export interface OracleResponse extends TransactionAttribute {
+    type: "OracleResponse";
+    id: number;
+    code: OracleResponseCode;
+    result?: Base64Encoded;
+}
+
+export type Transaction = {
+    hash: UInt256;
+    size: number;
+    blockHash: UInt256;
+    blockTime: number;
+    confirmations: number;
+    version: number;
+    nonce: number;
+    systemFee: Integer;
+    networkFee: Integer;
+    validUntilBlock: number;
+    sender: UInt160;
+    signers: Signer[];
+    attributes: TransactionAttribute[];
+    script: Base64Encoded;
+}
+
+export type Block = {
+    hash: UInt256;
+    size: number;
+    confirmations: number;
+    nextBlockHash?: UInt256;
+    version: number;
+    previousBlockHash: UInt256;
+    merkleRoot: UInt256;
+    time: number;
+    nonce: HexString;
+    index: number;
+    primary: number;
+    nextConsensus: UInt160;
+    tx: Transaction[];
+}
+
+export type TriggerType =
+    "OnPersist" |
+    "PostPersist" |
+    "Verification" |
+    "Application";
+
+export type VMState =
+    "NONE" |
+    "HALT" |
+    "FAULT" |
+    "BREAK";
+
+export type StackItemType =
+    "Any" |
+    "Pointer" |
+    "Boolean" |
+    "Integer" |
+    "ByteString" |
+    "Buffer" |
+    "Array" |
+    "Struct" |
+    "Map" |
+    "InteropInterface";
+
+export type Argument = {
+    type: StackItemType;
+    value?: any;
+}
+
+export type ApplicationLog = {
+    txid: UInt256;
+    executions: {
+        trigger: TriggerType;
+        vmstate: VMState;
+        exception?: string;
+        gasconsumed: Integer;
+        stack: Argument[];
+        notifications: Notification[];
+    }[];
+}
+
+export type Token = {
+    symbol: string;
+    decimals: number;
+    totalSupply: Integer;
+}
+
+// Provides the necessary arguments for a contract call.
+export type InvocationArguments = {
+    hash: UInt160;         // The hash of the contract to be called.
+    operation: string;     // The operation of the contract to be called.
+    args?: Argument[];     // The arguments for the call.
+    abortOnFail?: boolean; // Indicates whether the entire transaction should fail when the contract returns `false`.
+}
+
+export type InvocationResult = {
+    script: Base64Encoded;
+    state: VMState;
+    gasconsumed: Integer;
+    exception?: string;
+    stack: Argument[];
+}
+
+export type ContractParametersContext = {
+    type: "Neo.Network.P2P.Payloads.Transaction";
+    hash: UInt256;
+    data: Base64Encoded;
+    items: Record<UInt160, {
+        script: Base64Encoded;
+        parameters: {
+            type: ContractParameterType;
+            value: any;
+        }[];
+        signatures: Record<ECPoint, Base64Encoded>;
+    }>;
+    network: Network;
+}
+
+export type AuthenticationChallengePayload = {
+    action: "Authentication";
+    grant_type: "Signature";
+    allowed_algorithms: ["ECDSA-P256"];
+    network: Network;

```

Comment:
I think `network` could be an array, so the dapp could specify that it allows connecting with many networks and is up to the wallet to decide which of those it will connect. This gives more flexibility for dapp testing and if the dapp wants to avoid this complexity it could describe a single value.

Reply:
I'm not sure if this is possible. Because the address format of different networks may be different.

Reply:
Sure, but the account will be different for each connection. It's not supposed to let the wallet change the network on the fly, only to choose which network it will use on the entire connection, from beginning to end. I've already implemented such interaction between dapp and wallet on my Wallet Connect experiments and it makes sense.

Reply:
Done.